proposals/function-pointers.md

```diff
@@ -243,6 +243,26 @@ The better function member specification will be changed to include the followin
 
 This means that it is possible to overload on `void*` and a `delegate*` and still sensibly use the address-of operator.
 
+## Metadata representation of `in`, `out`, and `ref readonly` parameters and return types
```

Comment:
Do the rules for how the language converts a managed method signature to a function pointer signature need to be defined here?

That is, C# allows you to apply your own `[In]` and `[Out]` attributes (with certain limitations) today:
* `in` parameters and `ref readonly` return
  * Has the `IsReadOnly` attribute
  * Virtual methods additionally have a `modreq [In]` attribute
  * Sets the `in` built-in param attribute
  * Allows users to explicit specify `[In]`
  * Errors if the user specifies `[Out]`
* `out` parameters
  * Sets the `out` built-in param attribute
  * Allows users to explicitly specify `[Out]`
  * Errors if the user specifies `[In]`
* `ref` parameters and `ref` return
  * Allows users to explicitly specify `[In]`, which sets the `in` built-in param attribute
  * Allows users to explicitly specify `[In, Out]` which sets the `in` and `out` built-in param attributes
  * Errors if the user only specifies `[Out]`

I imagine the intent is it then behaves as the following when encountering a "ref-like" param in IL (`T&`):
* If it has the `IsReadOnly` attribute, the function-pointer signature will have `modreq [In]` for that parameter/return
* If it has the `out` built-in param attribute, the function pointer signature will have `modreq [Out]` for that parameter
* Otherwise, the function pointer signature will have no `modreq` for that parameter/return

Is that correct?

That is, it is my interpretation that it will ignore things which the language doesn't currently use to detect the "type"

Reply:
We don't need to specify anything here. The language understands method signatures today, and method-function pointer conversions are already defined based on that understanding.